### ISSUE TITLE
TST Wheel 'skipping' with fast scrolling

### ISSUE THREAD

**Issue by dxnull** [joshuacant/tst-wheel_and_double#18]
@joshuacant, thank you for creating this add-on, this is one feature I can't live without anymore when using TST, however I did find one unusual issue with it.

I've found that scrolling seems to be 'rate limited' in a certain way, as scrolling too quickly or using something like the free scrolling mode on my Logitech mouse will cause it to not switch between tabs at the same rate as the actual scroll events coming from the mouse. I do not believe the issue is related to Firefox or TST itself as I can use the following AHK script to reproduce similar functionality and it will scroll through the tabs at the proper rate:

```
#MaxHotkeysPerInterval 200
DoWeNeedToFireMouseWheel()
{
MouseGetPos, x1, y1, hw1
if (y1 >= 60 && x1 < 280 && WinActive("ahk_class MozillaWindowClass"))
return 1
}
#If, DoWeNeedToFireMouseWheel()
WheelUp::
Send ^+{Tab}
return
WheelDown::
Send ^{Tab}
return
#If
```

It's more of an annoyance than anything as it means that you need to force yourself to scroll slower than usual if you need to move through a large number of tabs.

The AHK script does have some downsides (hardcoded scrolling area, does not work if FF window is inactive), so hopefully this is an easy fix, otherwise I can just live with it. :)

Either way, thank you for this add-on and your hard work!

**Comment by joshuacant** [comment]
Yep, there's some code that was added by a contributor that limits the scrolling events to one per 100ms, or 10 times a second. I believe, based on the code comments, that this was added to prevent tabs snapping back and forth if you have a loose scroll wheel, or a very sensitive touch pad/sloppy fingers.

I'm inclined to leave that code in place, but I might add an option to bypass it in a future version. I'll leave this ticket open until I've either added that feature, or decided not to.

**Comment by dxnull** [comment]
> Yep, there's some code that was added by a contributor that limits the scrolling events to one per 100ms, or 10 times a second. I believe, based on the code comments, that this was added to prevent tabs snapping back and forth if you have a loose scroll wheel, or a very sensitive touch pad/sloppy fingers.
> 
> I'm inclined to leave that code in place, but I might add an option to bypass it in a future version. I'll leave this ticket open until I've either added that feature, or decided not to.

Ah, makes sense. If this could be exposed as a configurable option, that would be great. :)

Thanks!

**Comment by AllenAnthes** [comment]
@joshuacant I added the 100ms limit as part of the window scrolling feature, it doesn't actually effect scrolling through tabs.

I have the same issue when using something like freescroll, and assumed it was caused by some kind of debounce within TST or just the time it takes for the messages to go between the two addons.

**Comment by joshuacant** [comment]
Alright, so it sounds like this is... Maybe possible, but also maybe not possible due to limitations outside the scope of what we can do in this addon. I'm going to leave the ticket open and mark it as help wanted, but I don't think I'm going to go down this particular rabbit hole.

**Comment by amandamana** [comment]
@dxnull your AHK script is interacting directly with Firefox so you can't conclude that the issue is not related to TST itself. I think the speed is a limitation of the APIs and the issue should be closed as impossible to address within an addon.

**Comment by joshuacant** [comment]
Hi all, I have FINALLY (sorry!!!) uploaded a beta version to AMO with this change (and others). If you would like to try it out, visit this link: https://addons.mozilla.org/firefox/downloads/file/4181773/tree_style_tab_mouse_wheel-1.5.4.xpi

It's a fully signed extension by Mozilla, so you can install it in normal Firefox, but it's not pushed out via the main channel. So if you encounter any problems you can let me know and then downgrade to the published version 1.5 here: https://addons.mozilla.org/en-US/firefox/addon/tree-style-tab-mouse-wheel/

Feedback would be appreciated.

**Comment by joshuacant** [comment]
I've been informed that at least some people are getting an error on that addons.mozilla.org link. I uploaded the test XPI here: https://github.com/joshuacant/tst-wheel_and_double/releases/tag/1.5.4
